SYCL/Basic/parallel_for_user_types.cpp

```diff
@@ -0,0 +1,64 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
```

Comment:
```suggestion
// RUN: %HOST_RUN_PLACEHOLDER %t.out
// RUN: %CPU_RUN_PLACEHOLDER %t.out
// RUN: %GPU_RUN_PLACEHOLDER %t.out
// RUN: %ACC_RUN_PLACEHOLDER %t.out
```
We should run on all possible devices separately to verify all possible taret devices